src/python/tests/test_protocols.py

```diff
@@ -0,0 +1,121 @@
+################################################################################
+# MIT License
+#
+# Copyright (c) 2021 IBM
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+################################################################################
+"""ALog unit tests to confirm that the protocol is aligned `logging.Logger`."""
+
+import importlib
+import inspect
+import logging
+from typing import List
+
+from alog.protocols import LoggerProtocol
+
+
+def get_parameter_names(obj: object, method_name: str) -> List[str]:
+    """
+    Get the parameter names for an object's method
+    """
+    signature = inspect.signature(getattr(obj, method_name))
+    return list(signature.parameters)
+
+
+# Reset logging to its original definition
+importlib.reload(logging)
```

Comment:
I'm wondering if we need to orchestrate this carefully with the other tests? CI seems to be passing, so I think we're ok, but this seems like it might be something we want to put in a `@pytest.fixture` context manager with monkey patching `sys.modules` rather than using `reload` at import time for the tests.